okio/src/commonMain/kotlin/okio/Channel.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2025 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+/**
+ * A pair of streams for interactive communication with another machine.
+ *
+ * Send data to the peer by writing to [sink], and read data from the peer by reading from [source].
+ *
+ * This can be implemented by a plain TCP socket. It can also be layered to add features like
```

Comment:
The Socket (TCP, SSL) case is interesting, because I assume we'd want some of the features from https://github.com/square/okhttp/pull/8410/files#diff-e108767b1cd01f2dccec45ffa13f1463516eed6ebebf1c25e8bb961bc88922f8

```
  var soTimeout: Int
  val peerName: String
  val localPort: Int
  val inetAddress: InetAddress?
  val localAddress: InetAddress
  val remoteSocketAddress: SocketAddress?
  ```